src/emqx_ws_connection.erl

```diff
@@ -264,9 +264,9 @@ websocket_info(Info, State) ->
     ?WSLOG(error, "unexpected info: ~p", [Info], State),
     {ok, State}.
 
-terminate(SockError, _Req, State = #state{keepalive   = Keepalive,
-                                          proto_state = ProtoState,
-                                          shutdown    = Shutdown}) ->
+terminate(SockError, _Req, #state{keepalive   = Keepalive,
+                                  proto_state = ProtoState,
+                                  shutdown    = Shutdown}) ->
     ?WSLOG(debug, "Terminated for ~p, sockerror: ~p",
            [Shutdown, SockError], State),
```

Comment:
Perhaps not related to this PR. but the removal of `State` variable binding made me wonder why it compiled without error for this line.